src/runtime/actor/report.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{error::ActorError, Actor};
+
+/// An actor exit report
+pub type Report<A> = Result<SuccessReport<A>, ErrorReport<A>>;
+
+/// A report that an actor finished running with an error
+#[derive(Debug)]
+pub struct SuccessReport<A: Actor> {
+    /// The actor's state when it finished running
+    pub state: A,
+    /// The actor's data when it finished running
+    pub data: Option<A::Data>,
+}
```

Comment:
okay, that's a conflict with what I suggested earlier about renaming `data` to `state`. You may not like this but I would rename `state` here to `actor` or something similar, because that's exactly what the type `A` says is stored in there. I know what you want to express here, and it also makes sense, so I guess it comes down to preferences again. An alternative could then be:
```rust
#[derive(Debug)]
pub struct SuccessReport<A: Actor> {
    pub stopped_actor: A,
    pub final_state: Option<A::State>,
}
```
I know it's a lot more wordy, but it constantly reminds the reader in whatever file s/he's in what's relevent to know about those fields without consulting the docs or jumping to this module.

Same suggestion goes for the ErrorReport struct.

Reply:
Made some naming changes I felt were appropriate. Let me know if you still hate it.